WordPress/gutenberg
# `BoxControl`: Update story and refactor to Typescript

## What?


Via https://github.com/WordPress/gutenberg/pull/56185#discussion_r1397453288 it was found that `BoxControl`'s story hasn't been updated to use new storybook features. This required updating to Typescript as well.  

## Why?


For consistency and easier use of the docs. 

## How?


Refactoring to Typescript, following our storybook standards by adding a meta object, and using the template and args format.  

## Testing Instructions
1. `npm run: storybook dev`
2. Ensure `BoxControl` examples behave and look the same as before
3. See that expected storybook features are there (controls in docs)
